File: modified_scripts/r/merged_scripts/merged_matrix_multiplication.r

# qwen2_5-coder_3b CODE

mat_mult <- function(A, B) {
  nrow_A <- nrow(A)
  ncol_A <- ncol(A)
  nrow_B <- nrow(B)
  ncol_B <- ncol(B)
  
  if (ncol_A != nrow_B) {
    stop("Matrix dimensions do not allow multiplication.")
  }
  
  C <- matrix(0, nrow = nrow_A, ncol = ncol_B)
  
  for (i in 1:nrow_A) {
    for (j in 1:ncol_B) {
      for (k in 1:ncol_A) {
        C[i, j] <- C[i, j] + A[i, k] * B[k, j]
      }
    }
  }
  
  return(C)
}
# gemma_2b CODE

matrix_multiplication <- function(matrix1, matrix2) {
  n <- nrow(matrix1)
  m <- ncol(matrix2)
  result <- matrix(NA, n, m)
  for (i in 1:n) {
    for (j in 1:m) {
      result[i, j] <- matrix1[i, 1] * matrix2[j, 1] + matrix1[i, 2] * matrix2[j, 2] + ... + matrix1[i, n] * matrix2[j, n]
    }
  }
  return(result)
}
# qwen3_8b CODE
matrix_multiplication <- function(a, b) {
  n <- nrow(a)
  m <- ncol(a)
  p <- ncol(b)
  result <- matrix(0, n, p)
  for (i in 1:n) {
    for (j in 1:p) {
      for (k in 1:m) {
        result[i, j] <- result[i, j] + a[i, k] * b[k, j]
      }
    }
  }
  result
}
# deepseek-coder_1_3b CODE

matrix_multiplication = function(A, B) {
    if (nrow(A)!= ncol(B))  stop("Matrix dimensions are not correct")  
    result = rep(0, dim(A)[1]) 									    
    result = 	apply(A, MARGIN=2 , FUN = function (i){ 
        return(sum((B[i, ]*matrix_partitions)))  
    })            
    return(result)}   

# deepseek-coder_6_7b CODE

matrix_multiplication <- function(A, B) {
  if (ncol(A) != nrow(B)) stop("Matrices are not m*n and n*p")
  
  p = matrix(0, nrow(A), ncol(B))
  
  for (i in 1:nrow(A)) {
    for (j in 1:ncol(B)) {
      for (k in 1:ncol(A)) {
        p[i, j] = p[i, j] + A[i, k] * B[k, j]
      }
    }
  }
  
  return(p)
}
# qwen2_5-coder_7b CODE

matrix_multiplication <- function(A, B) {
  n <- nrow(A)
  m <- ncol(B)
  p <- ncol(A)
  
  C <- matrix(0, n, m)
  
  for (i in 1:n) {
    for (j in 1:m) {
      for (k in 1:p) {
        C[i, j] <- C[i, j] + A[i, k] * B[k, j]
      }
    }
  }
  
  return(C)
}
# gemma_7b CODE

matrix_multiplication <- function(matrix1, matrix2) {
  if (ncol(matrix1) != nrow(matrix2)) {
    stop("Matrices are not compatible for multiplication.")
  }
  result <- matrix(0, nrow(matrix1), ncol(matrix2))
  for (i in 1:nrow(matrix1)) {
    for (j in 1:ncol(matrix2)) {
      for (k in 1:ncol(matrix1)) {
        result[i, j] <- result[i, j] + matrix1[i, k] * matrix2[k, j]
      }
    }
  }
  return(result)
}
# qwen3_1_7b CODE

File: modified_scripts/rust/merged_scripts/merged_matrix_multiplication.rs

# gemma_7b CODE

fn matrix_multiplication(
    matrix1: Vec<Vec<f64>>,
    matrix2: Vec<Vec<f64>>,
) -> Vec<Vec<f64>> {
    let m1 = matrix1.len();
    let n1 = matrix1[0].len();
    let m2 = matrix2.len();
    let n2 = matrix2[0].len();

    if n1 != m2 {
        panic!("Matrices are not compatible for multiplication");
    }

    let mut result = vec![vec![0.0; n2]; m1];

    for i in 0..m1 {
        for j in 0..n2 {
            for k in 0..n1 {
                result[i][j] += matrix1[i][k] * matrix2[k][j];
            }
        }
    }

    result
}

# deepseek-coder_6_7b CODE

pub fn matrix_multiplication(a: &Vec<Vec<f64>>, b: &Vec<Vec<f64>>) -> Vec<Vec<f64>> {
    let a_rows = a.len();
    let a_cols = a[0].len();
    let b_cols = b[0].len();
    
    let mut result: Vec<Vec<f64>> = vec![vec![(0.0; b_cols)]; a_rows];
    
    for i in 0..a_rows {
        for j in 0..b_cols {
            for k in 0..a_cols {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    
    result
}
# qwen2_5-coder_3b CODE

fn matrix_multiply(a: &[[i32; 4]; 4], b: &[[i32; 4]; 4]) -> [[i32; 4]; 4] {
    let mut result = [[0; 4]; 4];
    for i in 0..4 {
        for j in 0..4 {
            for k in 0..4 {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    result
}

# deepseek-coder_1_3b CODE

pub fn matrix_multiplication(a: &[Vec<f32>], b: Vec<Vec<f32>>) -> Result<MatrixMultiplicationResult, String> {    
    let (rows1, cols0) = a.iter().fold((0usize, 0), |(count, count_inner), _| (count+1, *count_inner));   
                                                                                            if !b[0].is_empty() {        # Otherwise the number column mismatch.      return Err("Number Column Mismatched".to_string());  }     let cols1 = b[0].len();   // calculate rows in first matrix and total columns of second one respectively    match a[rows][*cols]*{       case [Some(x), Some(y)]: {         return Ok((a, x as f32 * y));        }                                   #  default:             break;      };     Err("Matrix Multiplication not possible".to_string())
# qwen3_1_7b CODE
fn matrix_mult(a: &[Vec<i32>, b: &[Vec<i32>]) -> Vec<Vec<i32> > {
    let rows_a = a.len();
    let cols_a = a[0].len();
    let rows_b = b.len();
    let cols_b = b[0].len();

    let mut result = vec![vec![]; rows_a];

    for i in 0..rows_a {
        for j in 0..cols_b {
            let mut val = 0;
            for k in 0..cols_a {
                val += a[i][k] * b[k][j];
            }
            result[i][j] = val;
        }
    }

    result
}
# qwen2_5-coder_7b CODE

fn matrix_multiplication(a: &Vec<Vec<i32>>, b: &Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let rows_a = a.len();
    let cols_a = a[0].len();
    let rows_b = b.len();
    let cols_b = b[0].len();

    assert_eq!(cols_a, rows_b);

    let mut result = vec![vec![0; cols_b]; rows_a];

    for i in 0..rows_a {
        for j in 0..cols_b {
            for k in 0..cols_a {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }

    result
}

# qwen3_8b CODE
fn matrix_mult(a: &[Vec<i32>], b: &[Vec<i32>]) -> Vec<Vec<i32>> {
    let rows = a.len();
    let cols = b[0].len();
    let mut result = vec![vec![0; cols]; rows];
    for i in 0..rows {
        for j in 0..cols {
            for k in 0..a[0].len() {
                result[i][j] += a[i][k] * b[k][j]
            }
        }
    }
    result
}
# gemma_2b CODE

fn matrix_multiplication(matrix1: [f32; 3], matrix2: [f32; 3]) -> [f32; 3] {
    
    let mut result_matrix = [0.0; 3];

    
    for i in 0..3 {
      
        for j in 0..3 {
      
            let product = matrix1[i] * matrix2[j];

      
            result_matrix[i][j] += product;
        }
    }

   
    result_matrix
}
